### DISCUSSION TITLE
Lesson 12 how to get WETH on Anvil chain

### DISCUSSION THREAD

**Discussion by LanceAddison** [Cyfrin/foundry-full-course-cu#1973] [Q&A] [answered]
I was able to deploy my contract to anvil but now I am not sure how to get weth to my anvil address. I got the collateralToken addresses using the ```getCollateralTokens``` function with this command ```cast call 0x0165878A594ca255338adfa4d48449f69242Eb8F "getCollateralTokens()(address[] memory)" --rpc-url $ANVIL_RPC_URL``` but when I passed the weth address I got to this command ```cast send 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "deposit()" --value 0.1ether --rpc-url $ANVIL_RPC_URL --private-key $DEFAULT_ANVIL_KEY``` I got this error ```server returned an error response: error code -32602: Invalid input: 'max_priority_fee_per_gas' greater than 'max_fee_per_gas'```. If anyone can help me with this I would appreciate it.

**Comment by EngrPips** [comment]
Hello @LanceAddison, were you running an `Anvil` chain in a second terminal when you ran this `cast send 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "deposit()" --value 0.1ether --rpc-url $ANVIL_RPC_URL --private-key $DEFAULT_ANVIL_KEY` command? You can also add `--legacy` flag to the command and retry running the command. Let's see if that works

**Reply by LanceAddison** [reply]
Yes, I had the ```Anvil``` chain in a second terminal. I also tried the ```--legacy``` flag but it didn't help. I assume the problem is with the ```ERC20Mock``` contract. In the ```HelperConfig.s.sol``` contract since I am using the anvil chain it deploys a new weth and wbtc contract using the ```ERC20Mock``` contract but in that contract there is no ```deposit()``` or ```withdraw()``` function like I see when I look at the weth contract on the sepolia testnet contract on etherscan. Do you know how to fix this? Could I simply add the functions from the weth contract on sepolia to my ```ERC20Mock``` contract and redeploy the contracts to ```Anvil``` or is there a contract my ```ERC20Mock``` contract can inherit from beside the ```ERC20``` contract from opnzeppelin to give me this functionality? Additionally, I pushed my project to my github so you can check out the code there if you want to.

**Reply by EngrPips** [reply]
You are currently doing the `Stable Coin` lesson, right? To be clear, are you trying to deposit ether to your `weth` contract?

**Reply by LanceAddison** [reply]
Yes, that is correct. I want to test my smart contract on Anvil. To do that I need to get ```weth``` to my anvil address since it only has eth by default. So yes I am pretty sure I am trying to deposit ether to get ```weth```. Also this address ```0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0``` that I pass to this command ```cast send 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "deposit()" --value 0.1ether --rpc-url $ANVIL_RPC_URL --private-key $DEFAULT_ANVIL_KEY``` is the valid mock weth contract that was deployed which the ```DSCEngine``` contract will allow collateral to be deposited since it has to be a valid token for the protocol to accept collateral just so you know.

**Reply by EngrPips** [reply]
Why don't you do all your testing in a test file Like Patrick did in the course? If you want to mint weth to an address, you call the `mint` function on the weth contract, and you don't need to deposit `ether` to get back `weth` token. The `weth` contract is just a simple `ERC20` contract where you can `mint,` `transfer,` and `burn` its token. You don't deposit into an `ERC20` contract.

**Reply by LanceAddison** [reply, marked as answer]
I did do all my testing in a test file. I just didn't want to deploy on ```sepolia``` yet and wanted to practice using ```cast``` in the terminal so I figured I would do it with ```anvil```. I figured out how to call the ```mint``` function for the ```weth``` contract using ```cast``` which is what I wanted. I just thought I had to deposit ```ether``` to get ```weth``` first. Thanks for the help.

**Reply by EngrPips** [reply]
You are welcome. Keep crushing it.
